clippy_lints/src/methods/unnecessary_fold.rs

```diff
@@ -40,6 +40,19 @@ fn needs_turbofish(cx: &LateContext<'_>, expr: &hir::Expr<'_>) -> bool {
         return false;
     }
 
+    // - the final expression in the body of a function with a simple return type
+    if let hir::Node::Block(block) = parent
+        && let grandparent = cx.tcx.parent_hir_node(block.hir_id)
+        && let hir::Node::Expr(grandparent_expr) = grandparent
+        && let ggparent = cx.tcx.parent_hir_node(grandparent_expr.hir_id)
```

Comment:
Probably better to iterate over the parents? is there any reason simply getting the containing item doesn't work?